src/server/gateways/postgres/plugin.ts

```diff
@@ -154,7 +158,7 @@ export default class PostgresGateway extends Gateway {
     const dd: String = String(today.getDate()).padStart(2, "0");
     const mm: String = String(today.getMonth() + 1).padStart(2, "0");
     const yyyy: String = String(today.getFullYear());
-    const date: string = `${mm}/${dd}/${yyyy}`;
```

Comment:
**Annotation:** Postgres results in an error when trying to insert dates in the format of mm/dd/yyyy. The format mm-dd-yyyy, however, is allowed by Postgres.

Reply:
Why is `generateDate` a public property of `PostgresGateway`? Why is it a function at all when we can generate the date much earlier in the request flow?

Reply:
Primarily because in the client side, users will not search for a computer generated time stamp. Instead they will search with mm/dd/yyy.

Reply:
Even then, we can still generate the date much earlier in the request flow; it's not clear why this function is categorically the responsibility of `PostgresGateway`.

Reply:
Relocate generate date method to controllers. Server is now responsible for generating a created_at for incoming documents.